index.js

```diff
@@ -1,21 +1,542 @@
+const DD_APPLICATION_KEY = "DD_APPLICATION_KEY";
+
 
+const sdkTestingDir = temp + "/.dd_sdk_testing_dir";
+const sdkTestingFrameworkPath = sdkTestingDir +"/dd_sdk_testing/DatadogSDKTesting.xcframework";
+const derivedDataPath = sdkTestingDir + "/derived";
+const xctestDir = derivedDataPath + "/Build/Products/";
+const testrunJson = sdkTestingDir + "/testrun.json";
+
+let envVars = Object.assign({}, process.env);
 
 // most @actions toolkit packages have async methods
 async function run() {
   try {
-    const ms = core.getInput('milliseconds');
-    core.info(`Waiting ${ms} milliseconds ...`);
+    let apiKey =  core.getInput("api_key");
+    if (apiKey) {
+      envVars[DD_API_KEY] = apiKey;
+    }
+
+    let applicationKey =  core.getInput("application_key");
+    if (applicationKey) {
+      envVars[DD_APPLICATION_KEY] = applicationKey;
+    }
+
+
+    let platform = core.getInput("platform") || "ios";
+    platform = platform.toLowerCase();
+
+    const sdk = core.getInput("sdk") || getSDKForPlatform(platform);
+    const destination = core.getInput("destination") || getDestinationForPlatform(platform);
+    const configuration = core.getInput("configuration") || "Debug";
+    const libraryVersion = core.getInput("libraryVersion") || "";
+    const extraParameters = core.getInput("extraParameters") || "";
+
+
+    //If project uses testplan force use of code coverage
+    let file_list = recFindByExt(".", "xctestplan");
+    for (let testPlanFile of file_list) {
+      await deleteLinesContaining(testPlanFile, "codeCoverage");
+    }
+
+    //Create folder to store files
+    if (!fs.existsSync(sdkTestingDir)) {
+      fs.mkdirSync(sdkTestingDir);
+    }
+
+    //Read project
+    const workspace = await getWorkspace();
+    let xcodeproj = await getXCodeProj();
+    var projectParameter;
+
+    //download testing framework
+    await downloadLatestFramework(libraryVersion);
+
+    if (workspace) {
+      console.log(`Workspace selected: ${workspace}`);
+      projectParameter = "-workspace " + `"${workspace}"`;
+    } else if (xcodeproj) {
+      console.log(`Project selected: ${xcodeproj}`);
+      projectParameter = "-project " + `"${xcodeproj}"`;
+    } else if (fs.existsSync("Package.swift")) {
+      console.log(`Package.swift selected`);
+      await swiftPackageRun(platform, extraParameters);
+      return;
+    } else {
+      core.setFailed(
+        "Unable to find workspace, project or Swift package file. Please set with workspace or xcodeproj"
+      );
+    }
+
+    const scheme = await getScheme(workspace, xcodeproj);
+    console.log(`Scheme selected: ${scheme}`);
+
+    //copy configfile
+    const configfileName = "ddTesting.xcconfig";
+    const configFilePath = sdkTestingDir + "/" + configfileName;
+    createXCConfigFile(configFilePath, sdkTestingDir);
+
+
+
+    let codeCoverParam = "-enableCodeCoverage YES";
+
+    //build for testing
+    const buildCommand = `xcodebuild build-for-testing ${codeCoverParam} ` +
```

Comment:
you're not using `codeCoverParam` or modifying later on, so we can set the literal directly: 
```suggestion
    //build for testing
    const buildCommand = `xcodebuild build-for-testing -enableCodeCoverage YES ` +
```